etrago/appl.py

```diff
@@ -133,7 +140,7 @@
         "n_segments": 5,
     },  # number of segments - only relevant for segmentation
     # Simplifications:
-    "skip_snapshots": 3,  # False or number of snapshots to skip
+    "skip_snapshots": 5,  # False or number of snapshots to skip
     "branch_capacity_factor": {"HV": 0.5, "eHV": 0.7},  # p.u. branch derating
     "load_shedding": False,  # meet the demand at value of loss load cost
     "foreign_lines": {
```

Comment:
When I set the foreign lines to 'DC' the pf_post_lopf fails: 
```
raceback (most recent call last):

  File "/home/clara/etrago_pypsa19/eTraGo/etrago/appl.py", line 494, in <module>
    etrago = run_etrago(args, json_path=None)

  File "/home/clara/etrago_pypsa19/eTraGo/etrago/appl.py", line 478, in run_etrago
    etrago.pf_post_lopf()

  File "/home/clara/etrago_pypsa19/eTraGo/etrago/tools/execute.py", line 271, in run_pf_post_lopf
    pf_post_lopf(self)

  File "/home/clara/etrago_pypsa19/eTraGo/etrago/tools/execute.py", line 455, in pf_post_lopf
    drop_foreign_components(network)

  File "/home/clara/etrago_pypsa19/eTraGo/etrago/tools/execute.py", line 354, in drop_foreign_components
    foreign_comp[comp].index]

  File "/home/clara/etrago_pypsa19/venv_pypsa19/lib/python3.7/site-packages/pandas/core/frame.py", line 3464, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1)[1]

  File "/home/clara/etrago_pypsa19/venv_pypsa19/lib/python3.7/site-packages/pandas/core/indexing.py", line 1314, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis)

  File "/home/clara/etrago_pypsa19/venv_pypsa19/lib/python3.7/site-packages/pandas/core/indexing.py", line 1377, in _validate_read_indexer
    raise KeyError(f"{not_found} not in index")

KeyError: "['47370 CH4', '47373 CH4', '47377 CH4', '47381 CH4', '47383 CH4', '47385 CH4', '47388 CH4', '47389 CH4', '47390 CH4', '47391 CH4', '47393 CH4', '47394 CH4', '47395 CH4', '47396 CH4', '47397 CH4', '47399 CH4', '47400 CH4', '47403 CH4', '47406 CH4', '47407 CH4', '47409 CH4', '47411 CH4', '47412 CH4', '47414 CH4', '47417 CH4', '47421 CH4', '47423 CH4', '47425 CH4', '47426 CH4', '47427 CH4', '47429 CH4', '47430 CH4', '47431 CH4', '47432 CH4', '47433 CH4', '47434 CH4', '47435 CH4', '47440 CH4', '47443 CH4', '47444 CH4', '47445 CH4', '47446 CH4', '47447 CH4', '47455 CH4', '47457 CH4', '47458 CH4', '47459 CH4', '47461 CH4', '47462 CH4', '47463 CH4', '47464 CH4', '47465 CH4', '47466 CH4', '47467 CH4', '47471 CH4', '47474 CH4', '47475 CH4', '47481 CH4', '47482 CH4', '47486 CH4', '47488 CH4', '47489 CH4', '47492 CH4', '47493 CH4', '47494 CH4', '47495 CH4', '47497 CH4', '47498 CH4', '47499 CH4', '47501 CH4', '47502 CH4', '47503 CH4', '47506 CH4', '47507 CH4', '47508 CH4', '47509 CH4', '47510 CH4', '47512 CH4', '47514 CH4', '47515 CH4', '47516 CH4', '47517 CH4', '47518 CH4', '47519 CH4', '47523 CH4', '47524 CH4', '47527 CH4', '47530 CH4', '47531 CH4', '47532 CH4', '47533 CH4', '47534 CH4', '47535 CH4', '47536 CH4', '47537 CH4', '47538 CH4', '47542 CH4', '47543 CH4', '47544 CH4', '47545 CH4', '47546 CH4', '47547 CH4', '47548 CH4', '47551 CH4', '47552 CH4', '47556 CH4', '47561 CH4', '47562 CH4', '47565 CH4', '47569 CH4', '47570 CH4', '47574 CH4', '47575 CH4', '47578 CH4', '47581 CH4', '47582 CH4', '47583 CH4', '47584 CH4', '47585 CH4', '47586 CH4', '47587 CH4', '47595 CH4', '47599 CH4', '47601 CH4', '47604 CH4', '47605 CH4', '47606 CH4', '47608 CH4', '47609 CH4', '47611 CH4', '47615 CH4', '47619 CH4', '47620 CH4', '47621 CH4', '47626 CH4', '47628 CH4', '47632 CH4', '47633 CH4', '47634 CH4', '47638 CH4', '47639 CH4', '47643 CH4', '47646 CH4', '47650 CH4', '47655 CH4', '47660 CH4', '47668 CH4', '47669 CH4', '47671 CH4', '47673 CH4', '47678 CH4', '47681 CH4', '47683 CH4', '47684 CH4', '47685 CH4', '47686 CH4', '47687 CH4', '47691 CH4', '47692 CH4', '47695 CH4', '47696 CH4', '47699 CH4', '47702 CH4', '47709 CH4', '47710 CH4', '47712 CH4', '47713 CH4', '47715 CH4', '47719 CH4', '47720 CH4', '47731 CH4', '47732 CH4', '47733 CH4', '47736 CH4', '47737 CH4', '47744 CH4', '47748 CH4', '47752 CH4', '47753 CH4', '47756 CH4', '47757 CH4', '47759 CH4', '47760 CH4', '47761 CH4', '47762 CH4', '47763 CH4', '47764 CH4', '47765 CH4', '47766 CH4', '47767 CH4', '47769 CH4', '47770 CH4', '47771 CH4', '47773 CH4', '47776 CH4', '47777 CH4', '
...
```

Was that working in your case?

Reply:
The function drop_foreign_components was updated to avoid this problem.

Reply:
Now it's working. Thank you!